compile/function.go

```diff
@@ -348,6 +365,33 @@ func (p *Parser) forIn() *ast.ForIn {
 	return &ast.ForIn{Var: ast.Ident{Name: id, Pos: pos}, E: expr, Body: body}
 }
 
+func (p *Parser) forSlice() *ast.For {
```

Comment:
"slice" makes sense in Go, but if we implement it in jSuneido that name won't make sense. Nor will it mean anything to Suneido programmers. I'd suggest something like `forCounted`

Reply:
*a slice of the number line?* :joy: 

but no, it's minor, will rename.